Text/CustomEntitySearch/CustomEntitySearch.cs

```diff
@@ -95,5 +132,174 @@ public static class CustomEntitySearch
+                        {
+                            minLevenshteinDistance[pointer] = distance;
+                            endofMatchInTextPointer[pointer] = nextPointer;
+                        }
+                        else if (distance > checkMatch.Length * 2)
+                            break;
+                    }
+                }
+                minLevenshteinDistance[pointersInText.Count - 1] = -1;
+
+                for (int i = 0; i < minLevenshteinDistance.Length; i++)
+                {
+                    if (minLevenshteinDistance[i] > -1 && minLevenshteinDistance[i] <= leniency)
+                    {
+                        entities.Add(
+                                new Entity
+                                {
+                                    Category = "customEntity",
+                                    Value = text.Substring(pointersInText[i], pointersInText[endofMatchInTextPointer[i]] - pointersInText[i] - 1),
+                                    Offset = pointersInText[i],
+                                    Confidence = minLevenshteinDistance[i]
+                                });
+                        entitiesFound.Add(checkMatch);
+                    }
+                }
+
+
+            }
+        }
+
+        private static double DamerauLevenshteinCalculation(string text, string checkMatch)
+        {
+            double[,] dynamicDistanceCalc = new double[text.Length + 1, checkMatch.Length + 1];
+            double substitutionCost = -1;
+            double accentAddition = 1;
+            for (int currTextIndex = 0; currTextIndex <= text.Length; currTextIndex++)
+                dynamicDistanceCalc[currTextIndex, 0] = currTextIndex;
+            for (int currWordIndex = 0; currWordIndex <= checkMatch.Length; currWordIndex++)
+                dynamicDistanceCalc[0, currWordIndex] = currWordIndex;
+
+            for (int currTextIndex = 0; currTextIndex < text.Length; currTextIndex++)
+            {
+                for (int currWordIndex = 0; currWordIndex < checkMatch.Length; currWordIndex++)
+                {
+                    if (text[currTextIndex].Equals(checkMatch[currWordIndex]))
+                        substitutionCost = 0;
+                    else if (String.Compare(checkMatch[currWordIndex].ToString(), text[currTextIndex].ToString(),
+                                    CultureInfo.CurrentCulture, CompareOptions.IgnoreNonSpace) == 0)
+                        substitutionCost = .5;
+                    else
+                        substitutionCost = 1;
+                    if (checkMatch[currWordIndex].IsAccent() ^ text[currTextIndex].IsAccent())
+                        accentAddition = 0.5;
+                    dynamicDistanceCalc[currTextIndex + 1, currWordIndex + 1] = Math.Min(
+                        Math.Min(dynamicDistanceCalc[currTextIndex, currWordIndex + 1] + accentAddition, 
```

Comment:
>dynamicDistanceCalc[ [](start = 33, length = 20)

nit: can you put each of these statements on a seperate line and add comments for which is substitution/addition/removal